### PR TITLE
Introduce vsync-based and overcommitment throttling to metal redrawer on macOS

### PR DESCRIPTION
**Pull request by elijah-semyonov**
Refactored version of https://github.com/JetBrains/skiko/pull/710

### CODE REVIEW

File: skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+@interface DisplayLinkThrottler : NSObject
+
+- (void)onVSync;
+
+@end
+
+static CVReturn displayLinkCallback(CVDisplayLinkRef displayLink, const CVTimeStamp *now, const CVTimeStamp *outputTime, CVOptionFlags flagsIn, CVOptionFlags *flagsOut, void *displayLinkContext) {
+    DisplayLinkThrottler *throttler = (__bridge DisplayLinkThrottler *)displayLinkContext;
+
+    [throttler onVSync];
```
Comment (igordmn) [inline skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm:18]: Will constant vsync affect battery on laptops?

**Review by igordmn — COMMENTED** [review]
A couple of comments, and it is good to merge, thanks!

File: skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm (hunk 1 above)
Comment (elijah-semyonov) [reply skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm:18]: I doubt so, even on iOS it's overhead is next to non-existant.

**Review by igordmn — APPROVED** [review]

**Comment by m-sasha** [comment]
Hurray!

**Pull request merged**